### PR TITLE
Fixing an error building with CUDA on windows

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

**Comment by Yangqing** [comment]
Actually, with #1993 this should not be needed any more. Should I land that diff?

**Comment by pjh5** [comment]
I think this may still be necessary for Windows. I still get the same error as before (lots of unresolved external symbols in cuda and cuDNN libraries) after rebasing onto #1993.

File: cmake/Utils.cmake [hunk 1]
```diff
@@ -244,7 +244,11 @@ function(caffe2_binary_target target_name_or_src)
   endif()
   add_executable(${__target} ${__srcs})
   add_dependencies(${__target} ${Caffe2_MAIN_LIBS_ORDER})
-  target_link_libraries(${__target} ${Caffe2_MAIN_LIBS} ${Caffe2_DEPENDENCY_LIBS})
+  if (USE_CUDA)
```
Comment (Yangqing) [inline cmake/Utils.cmake:247]: I assume this is no longer needed with the interface library we have?

Comment (pjh5) [reply cmake/Utils.cmake:247]: It actually was on Windows

Comment (Yangqing) [reply cmake/Utils.cmake:247]: nvm, saw your note above.

**Comment by Yangqing** [comment]
LGTM. Test failure inspected separately. Merging.

**Pull request merged**